README.md

```diff
@@ -1,165 +1,208 @@
+  - [▶️ Running bssd](#️-running-bssd)
+    - [🏁 Prerequisites:](#-prerequisites-3)
+  - [▶️ Running Network](#️-running-network)
+    - [🏁 Prerequisites:](#-prerequisites-4)
+    - [📚 Tutorial](#-tutorial-1)
+  - [🧪 Running Full Network Tests](#-running-full-network-tests)
+    - [🏁 Prerequisites:](#-prerequisites-5)
+    - [📚 Tutorial](#-tutorial-2)
+  - [📄 License](#-license)
 </details>
 
-## What is the Hemi Network?
+---
+## 🔧 Services
 
-Hemi is an EVM compatible L2 blockchain that brings Bitcoin security and Ethereum programability together.
+The Hemi Network comprises three key services, each serving a unique function within the network:
 
-### Services
+| Service | Description |
+|---------|-------------|
+| [**PoP Miner (popmd)**](https://github.com/hemilabs/heminetwork/blob/main/service/popm) | **Mines** L2 Keystones into Bitcoin blocks for Proof-of-Proof. |
+| [**Bitcoin Finality Governor (bfgd)**](https://github.com/hemilabs/heminetwork/blob/main/service/bfg) | Acts as the gateway to the Bitcoin network. |
+| [**Bitcoin Secure Sequencer (bssd)**](https://github.com/hemilabs/heminetwork/blob/main/service/bss) | Acts as a gateway to the Bitcoin Finality Governor (BFG) forked from Optimism, managing the consensus mechanisms of the Hemi Network.
 
-Hemi Network consists of 3 services:
 
-* [PoP Miner (popmd)](service/popm): "mines" L2 Keystones into BTC blocks for proof-of-proof
-* [Bitcoin Finality Governor (bfgd)](service/bfg): Hemi's gateway to the BTC network.
-* [Bitcoin Secure Sequencer (bssd)](service/bss): Optimism's gateway to BFG, manages Hemi Network's consensus
+---
+## 📜 TL;DR:
 
-### License
+- **extool**: A utility tool for extracting and processing data from various file formats, tailored for blockchain data analysis.
+- **hemictl**: The command-line interface for controlling and managing all Hemi Network services.
+- **keygen**: Generates and manages cryptographic keys used within the Hemi network, ensuring secure operations.
+- **popmd (Proof of Proof Miner Daemon)**: Embeds L2 Keystones into Bitcoin blocks for proof-of-proof, integral to the network's security.
+- **bfgd (Bitcoin Finality Governor Daemon)**: Manages connections and data transfers between the Hemi Network and the Bitcoin blockchain, ensuring finality.
+- **bssd (Bitcoin Secure Sequencer Daemon)**: Coordinates and sequences blockchain operations, serving as a bridge to the Bitcoin Finality Governor.
 
-This project is licensed under the [MIT License](LICENSE).
 
-## Getting Started
+---
 
-### Building from Source
+## ⚡️ Getting Started
 
-To build, you must have the following installed:
+### 🏁 Prerequisites
 
-* `git`
-* `make`
-* [Go v1.22+](https://go.dev/dl/)
+- `git`
+- `make`
+- [Go v1.22+](https://go.dev/dl/)
 
-First, clone the repository:
+---
 
-```shell
-git clone https://github.com/hemilabs/heminetwork.git
-```
+### 📚 Tutorial
 
-Then build:
 
-```shell
-cd heminetwork
-make deps
-make
-```
+1. Clone the Repository:
+   ```bash
```

Comment:
All commands in this file are shell commands and are not specific to the Bash shell, I think `shell` would be better for formatting.

Reply:
fixed